### ISSUE TITLE
enableCloudwatchLogsExports for DbCluster and DbInstance does not update the cloud resources

### ISSUE THREAD

**Issue by luis-alen** [aws-controllers-k8s/community#1845]
**Describe the bug**

When trying to update the attribute `enableCloudwatchLogsExports <[]string>` of an aurora-mysql DbCluster or a mysql DbInstance, the cloud resource is not updated/affected.

I've tried to enable the export of the slow query log for mysql according to the API reference (https://aws-controllers-k8s.github.io/community/reference/rds/v1alpha1/dbcluster/ and https://aws-controllers-k8s.github.io/community/reference/rds/v1alpha1/dbinstance/). 

```
spec:
  ...
  ...
  enableCloudwatchLogsExports:
    - slowquery
```

ACK reports the k8s object is in sync and that the property has been set.

```
Spec:
   ...
   ...

  Enable Cloudwatch Logs Exports:
    slowquery
  ...
  ...
  Conditions:
    Last Transition Time:     2023-07-13T13:23:33Z
    Message:                  Late initialization successful
    Reason:                   Late initialization successful
    Status:                   True
    Type:                     ACK.LateInitialized
    Last Transition Time:     2023-07-13T13:23:33Z
    Message:                  Resource synced successfully
    Reason:                   
    Status:                   True
    Type:                     ACK.ResourceSynced
```

**Steps to reproduce**

Create a mysql DbInstance or aurora-mysql DbCluster and try updating `enableCloudwatchLogsExports <[]string>` with

```
spec:
  ...
  ...
  enableCloudwatchLogsExports:
    - slowquery
```

**Expected outcome**

The actual cloud resource should get updated.

**Environment**

* Kubernetes version 1.23
* Using EKS (yes/no), if so version? NO
* AWS service targeted (S3, RDS, etc.) RDS
* ack-rds-controller version v.1.1.2

**Comment by a-hilaly** [comment]
@luis-alen I just checked the RDS code and it looks like we have update[ support cloudwatchlogs ](https://github.com/aws-controllers-k8s/rds-controller/blob/2272700afb6f5c29510e9ef79a0bd93bae75adbf/pkg/resource/db_cluster/custom_update.go#L398) - maybe we have a bug in the controller. Can you please provide some logs of the controller trying to handle this update?

**Comment by luis-alen** [comment]
Hello @a-hilaly. 

Yes, definitely. I've turned on debug log level and tried updating the cluster by setting the below attribute to its spec:

``` 
  enableCloudwatchLogsExports
    - slowquery
```

Here are the logs:

```
2023-07-26T18:15:13.645Z	DEBUG	ackrt	> r.Sync	{"account": "xxxxxxxxxx", "role": "", "region": "eu-central-1", "kind": "DBCluster", "namespace": "rc", "name": "dev-aurora2-mysql", "generation": 27}
2023-07-26T18:15:13.645Z	DEBUG	ackrt	>> r.resetConditions	{"account": "xxxxxxxxxx", "role": "", "region": "eu-central-1", "kind": "DBCluster", "namespace": "rc", "name": "dev-aurora2-mysql", "generation": 27}
2023-07-26T18:15:13.645Z	DEBUG	ackrt	<< r.resetConditions	{"account": "xxxxxxxxxx", "role": "", "region": "eu-central-1", "kind": "DBCluster", "namespace": "rc", "name": "dev-aurora2-mysql", "generation": 27}
2023-07-26T18:15:13.645Z	DEBUG	ackrt	>> rm.ResolveReferences	{"account": "xxxxxxxxxx", "role": "", "region": "eu-central-1", "kind": "DBCluster", "namespace": "rc", "name": "dev-aurora2-mysql", "is_adopted": true, "generation": 27}
2023-07-26T18:15:13.645Z	DEBUG	ackrt	<< rm.ResolveReferences	{"account": "xxxxxxxxxx", "role": "", "region": "eu-central-1", "kind": "DBCluster", "namespace": "rc", "name": "dev-aurora2-mysql", "is_adopted": true, "generation": 27}
2023-07-26T18:15:13.645Z	DEBUG	ackrt	>> rm.EnsureTags	{"account": "xxxxxxxxxx", "role": "", "region": "eu-central-1", "kind": "DBCluster", "namespace": "rc", "name": "dev-aurora2-mysql", "is_adopted": true, "generation": 27}
2023-07-26T18:15:13.645Z	DEBUG	ackrt	<< rm.EnsureTags	{"account": "xxxxxxxxxx", "role": "", "region": "eu-central-1", "kind": "DBCluster", "namespace": "rc", "name": "dev-aurora2-mysql", "is_adopted": true, "generation": 27}
2023-07-26T18:15:13.645Z	DEBUG	ackrt	>> rm.ReadOne	{"account": "xxxxxxxxxx", "role": "", "region": "eu-central-1", "kind": "DBCluster", "namespace": "rc", "name": "dev-aurora2-mysql", "is_adopted": true, "generation": 27}
2023-07-26T18:15:13.645Z	DEBUG	ackrt	>>> rm.sdkFind	{"account": "xxxxxxxxxx", "role": "", "region": "eu-central-1", "kind": "DBCluster", "namespace": "rc", "name": "dev-aurora2-mysql", "is_adopted": true, "generation": 27}
2023-07-26T18:15:13.747Z	DEBUG	ackrt	<<< rm.sdkFind	{"account": "xxxxxxxxxx", "role": "", "region": "eu-central-1", "kind": "DBCluster", "namespace": "rc", "name": "dev-aurora2-mysql", "is_adopted": true, "generation": 27}
2023-07-26T18:15:13.747Z	DEBUG	ackrt	<< rm.ReadOne	{"account": "xxxxxxxxxx", "role": "", "region": "eu-central-1", "kind": "DBCluster", "namespace": "rc", "name": "dev-aurora2-mysql", "is_adopted": true, "generation": 27}
2023-07-26T18:15:13.747Z	DEBUG	ackrt	>> r.updateResource	{"account": "xxxxxxxxxx", "role": "", "region": "eu-central-1", "kind": "DBCluster", "namespace": "rc", "name": "dev-aurora2-mysql", "is_adopted": true, "generation": 27}
2023-07-26T18:15:13.747Z	DEBUG	ackrt	<< r.updateResource	{"account": "xxxxxxxxxx", "role": "", "region": "eu-central-1", "kind": "DBCluster", "namespace": "rc", "name": "dev-aurora2-mysql", "is_adopted": true, "generation": 27}
2023-07-26T18:15:13.747Z	DEBUG	ackrt	>> r.lateInitializeResource	{"account": "xxxxxxxxxx", "role": "", "region": "eu-central-1", "kind": "DBCluster", "namespace": "rc", "name": "dev-aurora2-mysql", "is_adopted": true, "generation": 27}
2023-07-26T18:15:13.747Z	DEBUG	ackrt	>>> rm.LateInitialize	{"account": "xxxxxxxxxx", "role": "", "region": "eu-central-1", "kind": "DBCluster", "namespace": "rc", "name": "dev-aurora2-mysql", "is_adopted": true, "generation": 27}
2023-07-26T18:15:13.747Z	DEBUG	ackrt	no late initialization required.	{"account": "xxxxxxxxxx", "role": "", "region": "eu-central-1", "kind": "DBCluster", "namespace": "rc", "name": "dev-aurora2-mysql", "is_adopted": true, "generation": 27}
2023-07-26T18:15:13.747Z	DEBUG	ackrt	<<< rm.LateInitialize	{"account": "xxxxxxxxxx", "role": "", "region": "eu-central-1", "kind": "DBCluster", "namespace": "rc", "name": "dev-aurora2-mysql", "is_adopted": true, "generation": 27}
2023-07-26T18:15:13.747Z	DEBUG	ackrt	>>> r.patchResourceMetadataAndSpec	{"account": "xxxxxxxxxx", "role": "", "region": "eu-central-1", "kind": "DBCluster", "namespace": "rc", "name": "dev-aurora2-mysql", "is_adopted": true, "generation": 27}
2023-07-26T18:15:13.748Z	DEBUG	ackrt	no difference found between metadata and spec for desired and latest object.	{"account": "xxxxxxxxxx", "role": "", "region": "eu-central-1", "kind": "DBCluster", "namespace": "rc", "name": "dev-aurora2-mysql", "is_adopted": true, "generation": 27}
2023-07-26T18:15:13.748Z	DEBUG	ackrt	<<< r.patchResourceMetadataAndSpec	{"account": "xxxxxxxxxx", "role": "", "region": "eu-central-1", "kind": "DBCluster", "namespace": "rc", "name": "dev-aurora2-mysql", "is_adopted": true, "generation": 27}
2023-07-26T18:15:13.748Z	DEBUG	ackrt	<< r.lateInitializeResource	{"account": "xxxxxxxxxx", "role": "", "region": "eu-central-1", "kind": "DBCluster", "namespace": "rc", "name": "dev-aurora2-mysql", "is_adopted": true, "generation": 27}
2023-07-26T18:15:13.748Z	DEBUG	ackrt	>> r.ensureConditions	{"account": "xxxxxxxxxx", "role": "", "region": "eu-central-1", "kind": "DBCluster", "namespace": "rc", "name": "dev-aurora2-mysql", "is_adopted": true, "generation": 27}
2023-07-26T18:15:13.748Z	DEBUG	ackrt	<< r.ensureConditions	{"account": "xxxxxxxxxx", "role": "", "region": "eu-central-1", "kind": "DBCluster", "namespace": "rc", "name": "dev-aurora2-mysql", "is_adopted": true, "generation": 27}
2023-07-26T18:15:13.748Z	DEBUG	ackrt	< r.Sync	{"account": "xxxxxxxxxx", "role": "", "region": "eu-central-1", "kind": "DBCluster", "namespace": "rc", "name": "dev-aurora2-mysql", "is_adopted": true, "generation": 27}
2023-07-26T18:15:13.748Z	DEBUG	ackrt	requeuing	{"account": "xxxxxxxxxx", "role": "", "region": "eu-central-1", "kind": "DBCluster", "namespace": "rc", "name": "dev-aurora2-mysql", "is_adopted": true, "generation": 27, "after": "10m0s"}
2023-07-26T18:15:13.748Z	DEBUG	ackrt	> r.patchResourceStatus	{"account": "xxxxxxxxxx", "role": "", "region": "eu-central-1", "kind": "DBCluster", "namespace": "rc", "name": "dev-aurora2-mysql", "is_adopted": true, "generation": 27}
2023-07-26T18:15:13.748Z	DEBUG	ackrt	>> kc.Patch (status)	{"account": "xxxxxxxxxx", "role": "", "region": "eu-central-1", "kind": "DBCluster", "namespace": "rc", "name": "dev-aurora2-mysql", "is_adopted": true, "generation": 27}
2023-07-26T18:15:13.782Z	DEBUG	ackrt	patched resource status	{"account": "xxxxxxxxxx", "role": "", "region": "eu-central-1", "kind": "DBCluster", "namespace": "rc", "name": "dev-aurora2-mysql", "is_adopted": true, "generation": 27, "json": "{\"metadata\":{\"resourceVersion\":\"602179930\"},\"status\":{\"conditions\":[{\"lastTransitionTime\":\"2023-07-26T18:15:13Z\",\"status\":\"True\",\"type\":\"ACK.ResourceSynced\"}]}}"}
2023-07-26T18:15:13.782Z	DEBUG	ackrt	<< kc.Patch (status)	{"account": "xxxxxxxxxx", "role": "", "region": "eu-central-1", "kind": "DBCluster", "namespace": "rc", "name": "dev-aurora2-mysql", "is_adopted": true, "generation": 27}
2023-07-26T18:15:13.782Z	DEBUG	ackrt	< r.patchResourceStatus	{"account": "xxxxxxxxxx", "role": "", "region": "eu-central-1", "kind": "DBCluster", "namespace": "rc", "name": "dev-aurora2-mysql", "is_adopted": true, "generation": 27}
2023-07-26T18:15:13.782Z	DEBUG	ackrt	requeueing	{"account": "xxxxxxxxxx", "role": "", "region": "eu-central-1", "kind": "DBCluster", "namespace": "rc", "name": "dev-aurora2-mysql", "is_adopted": true, "generation": 27, "after": "10m0s"}
```

I couldn't spot any errors, but from this piece I believe it doesn't even recognize the change:

```
2023-07-26T18:15:13.748Z	DEBUG	ackrt	no difference found between metadata and spec for desired and latest object.	{"account": "xxxxxxxxxx", "role": "", "region": "eu-central-1", "kind": "DBCluster", "namespace": "rc", "name": "dev-aurora2-mysql", "is_adopted": true, "generation": 27}
```

**Comment by a10pepo** [comment]
I have faced the same issue in the update of the dbInstance, raised this PR as inspiration as I am not an expert in this terms:

https://github.com/aws-controllers-k8s/rds-controller/pull/157

**Comment by joaquinservetto** [comment]
> Hi, I'm also facing this issue. @a-hilaly is there any update? did you have the chance to review the PR from @a10pepo?

I noticed that the fix has to be done at the dbcluster level, I'm working on it, could you assign this issue to me? Thanks in advance @a-hilaly

**Comment by a-hilaly** [comment]
@joaquinservetto you can assign this issue by commenting `/assign` here:) . will do a review soon!

**Comment by joaquinservetto** [comment]
@a-hilaly, please take a look to this [PR](https://github.com/aws-controllers-k8s/rds-controller/pull/161) when you have some time, thanks!

**Comment by a-hilaly** [comment]
A fix will be included in rds controller 1.1.10 - thanks to @joaquinservetto contribution!
